### PR TITLE
added description to some functions 

### PR DESCRIPTION
**Pull request by i-redbyte**

### CODE REVIEW

**Review by siriak — DISMISSED** [review]

File: math/factorial/factorial.go [hunk 1]
```diff
@@ -24,6 +26,7 @@ func RecursiveFactorial(n int) int {
 	}
 }
 
+// CalculateFactorialUseTree This function finds the factorial of a number using a binary tree
 func CalculateFactorialUseTree(n int) int {
```
Comment (tjgurwara99) [inline math/factorial/factorial.go:30]:
```suggestion
// UsingTree computes and returns the factorial of n using a binary tree
func UsingTree(n int) int {
```

File: math/max/bitwisemax.go [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
 package max
 
+// BitwiseMax This function returns the maximum integer using bit operations
 func BitwiseMax(a int, b int, base int) int {
```
Comment (tjgurwara99) [inline math/max/bitwisemax.go:11]: `Bitwise` - max is in the name of package.

```suggestion
// Bitwise computes using bitwise operator the maximum of all the integer input and returns it
func Bitwise(a int, b int, base int) int {
```

File: math/min/bitwisemin.go [hunk 3]
```diff
@@ -7,6 +7,7 @@
 
 package min
 
+// Bitwise This function returns the minimum integer using bit operations
```
Comment (tjgurwara99) [inline math/min/bitwisemin.go:10]:
```suggestion
// Bitwise computes using bitwise operator the minimum of all the integer input and returns it
```

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

File: math/binary/xorsearch.go [hunk 4]
```diff
@@ -7,6 +7,7 @@
 
 package binary
 
+// XorSearchMissingNumber - This function find a missing number in a sequence
```
[the remark this answers is not in the document]
Comment (i-redbyte) [reply math/binary/xorsearch.go:10]: Done

File: math/binary/arithmeticmean.go [hunk 5]
```diff
@@ -8,10 +8,12 @@
 // Package binary describes algorithms that use binary operations for different calculations.
 package binary
 
+// MeanUsingAndXor This function find arithmetic mean using "AND" and "XOR" operations
```
[the remark this answers is not in the document]
Comment (i-redbyte) [reply math/binary/arithmeticmean.go:11]: Done

File: math/binary/arithmeticmean.go [hunk 6]
```diff
@@ -8,10 +8,12 @@
...
 func MeanUsingAndXor(a int, b int) int {
 	return ((a ^ b) >> 1) + (a & b)
 }
 
+// MeanUsingRightShift This function find arithmetic mean using right shift
```
[the remark this answers is not in the document]
Comment (i-redbyte) [reply math/binary/arithmeticmean.go:16]: Done

File: math/catalan/catalannumber.go [hunk 7]
```diff
@@ -12,6 +12,7 @@ import (
 	f "github.com/TheAlgorithms/Go/math/factorial"
 )
 
+//CatalanNumber This function returns the Catalan number by its number
```
[the remark this answers is not in the document]
Comment (i-redbyte) [reply math/catalan/catalannumber.go:15]: Done

File: math/factorial/factorial.go [hunk 8]
```diff
@@ -8,6 +8,7 @@
 // Package factorial describes algorithms Factorials calculations.
 package factorial
 
+// BruteForceFactorial This function calculates the factorial of a number using a loop
```
[the remark this answers is not in the document]
Comment (i-redbyte) [reply math/factorial/factorial.go:11]: Done

File: math/factorial/factorial.go [hunk 9]
```diff
@@ -8,6 +8,7 @@
...
 func BruteForceFactorial(n int) int {
```
[the remark this answers is not in the document]
Comment (i-redbyte) [reply math/factorial/factorial.go:12]: Done

File: math/factorial/factorial.go [hunk 10]
```diff
@@ -16,6 +17,7 @@ func BruteForceFactorial(n int) int {
 	return result
 }
 
+// RecursiveFactorial This function recursively computes the factorial of a number
```
[the remark this answers is not in the document]
Comment (i-redbyte) [reply math/factorial/factorial.go:20]: Done

File: math/factorial/factorial.go (hunk 1 above)
Comment (i-redbyte) [reply math/factorial/factorial.go:30]: Done

File: math/max/bitwisemax.go (hunk 2 above)
Comment (i-redbyte) [reply math/max/bitwisemax.go:11]: Done

File: math/min/bitwisemin.go (hunk 3 above)
Comment (i-redbyte) [reply math/min/bitwisemin.go:10]: Done

**Review by siriak — DISMISSED** [review]

**Review by tjgurwara99 — DISMISSED** [review]

**Comment by raklaptudirm** [comment]
@i-redbyte Please resolve the merge conflicts.

**Comment by i-redbyte** [comment]
> @i-redbyte Please resolve the merge conflicts.

Done

**Comment by raklaptudirm** [comment]
@tjgurwara99 The workflow seems to be stuck again.

**Comment by tjgurwara99** [comment]
The workflow has been disabled until my last pr gets merged. It should not have run. Is this fork up to date with upstream?

**Comment by raklaptudirm** [comment]
Seems to be up to date.

**Pull request closed without merging**